src/transaction/locator_sr.c

```diff
@@ -14281,7 +14281,9 @@ locator_repl_start_tran (THREAD_ENTRY * thread_p)
   applier_Client_credentials.program_name = "(repl_applier)";
   applier_Client_credentials.process_id = -1;
 
-  int client_lock_wait = 0;
+  /* TODO : configurable lock wait for replication 
+   * zero time is not correct, since applier transaction may attempt to latch pages being currently vacuumed */
+  int client_lock_wait = TRAN_LOCK_INFINITE_WAIT;
```

Comment:
Is patch latch timeout the only problematic case? What about locking tables (for instance SCH_M_LOCK for DDL statements)?

Reply:
I don't think object lock matters. applier threads should be compatible with each other and ddl_proxy should be the sole writer of the database.